client/extensions/woocommerce/app/dashboard/widgets/inventory-widget/index.js

```diff
@@ -0,0 +1,185 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component, Fragment } from 'react';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { filter, sortBy } from 'lodash';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+// import Button from 'components/button';
+import {
+	areSettingsProductsLoaded,
+	getProductsSettingValue,
+} from 'woocommerce/state/sites/settings/products/selectors';
+import DashboardWidget from 'woocommerce/components/dashboard-widget';
+import { fetchProducts } from 'woocommerce/state/sites/products/actions';
+import { fetchSettingsProducts } from 'woocommerce/state/sites/settings/products/actions';
+import { getAllProducts } from 'woocommerce/state/sites/products/selectors';
+import { getLink } from 'woocommerce/lib/nav-utils';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import Table from 'woocommerce/components/table';
+import TableRow from 'woocommerce/components/table/table-row';
+import TableItem from 'woocommerce/components/table/table-item';
+
+class InventoryWidget extends Component {
+	static propTypes = {
+		isLoaded: PropTypes.bool.isRequired,
+		isLowStockEnabled: PropTypes.bool.isRequired,
+		lowStockThreshold: PropTypes.number.isRequired,
+		site: PropTypes.shape( {
+			ID: PropTypes.number.isRequired,
+		} ),
+		width: PropTypes.oneOf( [ 'half', 'full' ] ).isRequired,
+	};
+
+	componentDidMount() {
+		const { site } = this.props;
+		if ( site && site.ID ) {
+			this.props.fetchSettingsProducts( site.ID );
+			// Get all products
+			this.props.fetchProducts( site.ID, { offset: 0 } );
```

Comment:
So it appears that `fetchProducts` utilizes the [DEFAULT_QUERY](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/sites/products/utils.js#L9) which limits this request to 10 products per page. Wondering if we should consider bumping up that per page style for this widget to try and cast a wider net and get more products that might be low inventory / out of stock.

It looks like the [products endpoint](http://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products) supports an `in_stock` query arg too - so i suppose another route we could do here is if `props. isLowStockEnabled` is falsey - we just change this query to only grab out of stock products for display.

Reply:
The `offset: 0` triggers [a loop of product-fetches](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/data-layer/products/index.js#L138-L149), so the entire list of products is retrieved - but increasing that default per_page would mean fewer api requests so I'll do that.